pkg/image/importer/credentials.go

```diff
@@ -102,14 +102,15 @@ func NewLazyCredentialsForSecrets(secretsFn func() ([]kapi.Secret, error)) *Secr
 
 type SecretCredentialStore struct {
 	lock      sync.Mutex
+	origin    *url.URL
```

Comment:
This should be a `map[string /* origin */]string /* realm */`.
With the single value, the `origin` may be overwritten by other concurrent auth routines.